src/enums.h

```diff
@@ -361,6 +361,20 @@ struct game_message_params {
     game_message_flags flags;
 };
 
+enum class reachability_cache_quadrant : int {
+    NE, SE, NW, SW
+};
+
+template<>
+struct enum_traits<reachability_cache_quadrant> {
+    static constexpr reachability_cache_quadrant last = reachability_cache_quadrant::SW;
+    static constexpr int size = static_cast<int>( last ) + 1;
+
+    inline static reachability_cache_quadrant quadrant( bool S, bool W ) {
+        return static_cast<reachability_cache_quadrant>( ( W << 1 ) | S );
```

Comment:
The Visual Studio is throwing lot warnings. [C4805](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4805?view=msvc-160) 'operation' : unsafe mix of type 'int' and type 'bool' in operation. Can you refactor this?

Reply:
I think wrapping `S` and `W` in `static_cast<int>` should silence this warning. Since this is already merged, a separate PR is needed to add this.

Would you like to make it?

Reply:
Ok i will remake it. `static_cast<int>` will help.

Reply:
Thank you